pkg/metric/middleware.go

```diff
@@ -0,0 +1,30 @@
+package metric
+
+import (
+	"net/http"
+
+	klog "k8s.io/klog/v2"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// prometheusMiddleware implements mux.MiddlewareFunc.
+func PrometheusMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		route := mux.CurrentRoute(r)
+		path, _ := route.GetPathTemplate()
+		timer := prometheus.NewTimer(HttpDuration.WithLabelValues(path, r.Method))
+		klog.Infof("xxxxxxxxxxxxxxxx URL Referrer: %s", r.Referer())
+		klog.Infof("xxxxxxxxxxxxxxxx User Agent: %s", r.UserAgent())
+		next.ServeHTTP(w, r)
+		timer.ObserveDuration()
```

Comment:
Will this timer resolve when the request is completed?

Reply:
This timer does not work as expected. It only captures the time spent in this middleware. So removed that for now. Will try to catch response time using the other library that you sent. It works in a very different way BTW